explainers/anti-tracking.md

```diff
@@ -0,0 +1,247 @@
+navigations.
+
+#### Mark packages with a scheme
+
+```html
+<a href="package://package">
+```
+
+Basically encodes the `ispackage` flag into the scheme of the URL. The package
+is fetched by changing the scheme to `https`. We're [developing a more general
+package scheme](https://lists.w3.org/Archives/Public/uri/2019Nov/0000.html), but
+as it's useful for unsigned packages, which don't necessarily need the same
+restrictions as signed package, it seems like the scheme won't do the right
+thing.
+
+## Blocking normal link decoration
+
+Same-origin links to packages should use the same restrictions as cross-origin
+links to non-packages.
+
+## Blocking privileged link decoration
+
+A link source run by the same entity as the distributor can encode a user ID in
+ways that might break an arbitrary publisher's serving code. So, we need to
+define a deterministic function from the inner URL to the acceptable URL within
+a distributor used to fetch that inner URL's package.
+
+The two main options are
+
+```url
+https://distributor.origin/.well-known/package/<hash(innerUrl)>
+https://distributor.origin/.well-known/package/innerAuthority/innerPath?innerQuery
+```
+
+The hash-based path is a bit easier to compute, but leaving the inner URL mostly
+intact makes it easier for a human to guess where the link is going to go.
+
+## Privileged server-side collaboration
+
+The primary mechanism by which a distributor's server could collaborate with a
+link source's server is by deciding that if there's a click by user `A` at time
+`T` on a link to `https://distributor.example/package`, then a subsequent
+uncredentialed load of that package at time `T+ε` is probably user `A`. Servers
+can optionally refine this by limiting it to clicks and loads from the same IP
+address. This same mechanism allows cross-origin user ID transfer between
+arbitrary servers, and we don't know any ways to block it in that context, so we
+only need to make the package-assisted transfer at least as difficult as the
+cross-server transfer.
+
+We know of two ways for a distributor to pass their user ID onto the publisher
+of a package, and in either case—the publisher shares their signing key with the
+distributor, or the publisher generates 2^N variants of a package to encode N
+bits of a user ID—it seems safer or simpler for the publisher to share their
+logs instead.
+
+## Appendices
+
+### Security concerns
+
+There are worries that uncredentialed navigation could be used in an attack on
+the target site, since the main request won't send or save cookies, but
+subrequests will. We don't actually know a concrete attack that's enabled by
+uncredentialed navigations, but we'd like to prevent surprises anyway.
+
+Similar worries came up in the discussion of cross-origin prefetch, which led to
+the new `"prenavigate"` operation, for which the target site opts in using an
+[`Allow-Uncredentialed-Navigation`
+header](https://github.com/w3c/resource-hints/issues/82#issuecomment-529951528).
```

Comment:
Something that might be worth mentioning: SXGs ensures the credentialed and uncredentialed versions of the top-level resources are the same, which motivates for the package-only options.

Reply:
They don't guarantee they're the same. They basically allow the site to assert that the uncredentialed version is good enough, which is similar to the assertion with the `Allow-Uncredentialed-Navigation` header.

Reply:
Ah yeah, that's a better phrasing.